lib/srv/desktop/discovery.go

```diff
@@ -194,16 +194,19 @@ func (s *WindowsService) lookupDesktop(ctx context.Context, hostname string) (ad
 		return addrs, nil
 	}
 	if s.dnsResolver == nil {
-		return nil, trace.NewAggregate(err, trace.Errorf("DNS lookup for %v failed and there's no LDAP server to fallback to", hostname))
+		return nil, trace.NewAggregate(err, trace.Errorf("DNS lookup for %q failed and there's no LDAP server to fallback to", hostname))
 	}
-	s.cfg.Log.WithError(err).Debugf("DNS lookup for %v failed, falling back to LDAP server", hostname)
+	s.cfg.Log.WithError(err).Debugf("DNS lookup for %q failed, falling back to LDAP server", hostname)
 	return s.dnsResolver.LookupHost(ctx, hostname)
 }
 
 // ldapEntryToWindowsDesktop generates the Windows Desktop resource
 // from an LDAP search result
 func (s *WindowsService) ldapEntryToWindowsDesktop(ctx context.Context, entry *ldap.Entry, getHostLabels func(string) map[string]string) (types.ResourceWithLabels, error) {
 	hostname := entry.GetAttributeValue(windows.AttrDNSHostName)
+	if hostname == "" {
+		return nil, trace.BadParameter("missing hostname, attributes: %v", entry.Attributes)
```

Comment:
```suggestion
		return nil, trace.BadParameter("LDAP entry missing attribute %v, only found attributes: %v", windows.AttrDNSHostName, entry.Attributes)
```

Reply:
These errors during discovery don't ever make their way to users, they just get logged, so I think I prefer the brevity.